lib/livebook_web/helpers.ex

```diff
@@ -86,4 +86,30 @@ defmodule LivebookWeb.Helpers do
     pid_str = Phoenix.LiveDashboard.Helpers.encode_pid(pid)
     Routes.live_dashboard_path(socket, :page, node(), "processes", info: pid_str)
   end
+
+  @doc """
+  Converts human-readable strings to strings which can be used
+  as HTML element IDs (compatible with HTML5)
+
+  At the same time duplicate IDs are enumerated to avoid duplicates
+  """
+  @spec names_to_html_ids(Enum.t()) :: Stream.t()
+  def names_to_html_ids(names) do
+    names
+    |> Stream.map(&name_to_html_id/1)
+    |> Stream.transform(%{}, fn html_id, counts ->
+      counts = Map.update(counts, html_id, 1, &(&1 + 1))
+
+      case counts[html_id] do
+        1 -> {[html_id], counts}
+        count -> {["#{html_id}-#{count}"], counts}
+      end
+    end)
+  end
+
+  defp name_to_html_id(name) do
+    name
+    |> String.trim()
+    |> String.replace(~r/\s+/u, "-")
```

Comment:
I think it would still make sense to downcase the result.

Reply:
Sure, I wondered about that 😄   Done!